libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt

```diff
@@ -97,16 +108,36 @@ class MapboxOnboardRouter : Router {
             )
         ).build()
 
-        OfflineRouteRetrievalTask(navigatorNative, logger, object : OnOfflineRouteFoundCallback {
-            override fun onRouteFound(routes: List<DirectionsRoute>) {
-                callback.onResponse(routes)
+        retrieveRoute(offlineRouter.buildUrl(), callback)
+    }
+
+    override fun cancel() {
+        jobControl.scope.cancel()
+    }
+
+    private fun retrieveRoute(url: String, callback: Router.Callback) {
+        jobControl.scope.launch {
+            val routerResult = withContext(Dispatchers.Default) {
```

Comment:
Should we use the dispatcher from `ThreadController` instead?

Reply:
done